cmd/andromeda-liquid-server/main.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/sapcc/go-api-declarations/liquid"
+	"github.com/sapcc/go-bits/httpext"
+	"github.com/sapcc/go-bits/liquidapi"
+	"github.com/sapcc/go-bits/must"
+	"github.com/urfave/cli/v2"
+
+	"github.com/sapcc/andromeda/client"
+	"github.com/sapcc/andromeda/client/administrative"
+	"github.com/sapcc/andromeda/models"
+)
+
+type liquidLogic struct {
+	andromedaClient *client.Andromeda
+}
+
+func (l *liquidLogic) Init(_ context.Context, provider *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) error {
+	l.andromedaClient = client.Default
+	endpointOpts := gophercloud.EndpointOpts{
+		Region: eo.Region,
+	}
+	endpointOpts.ApplyDefaults("gtm")
+	endpoint, err := provider.EndpointLocator(endpointOpts)
+	if err != nil {
+		return err
+	}
+	uri, err := url.Parse(endpoint)
+	if err != nil {
+		return err
+	}
+	rt := runtimeclient.New(uri.Host, uri.Path, []string{uri.Scheme})
+	rt.DefaultAuthentication = runtime.ClientAuthInfoWriterFunc(func(req runtime.ClientRequest, reg strfmt.Registry) error {
+		if err := req.SetHeaderParam("X-Auth-Token", provider.Token()); err != nil {
+			return err
+		}
+		return nil
+	})
+	l.andromedaClient.SetTransport(rt)
+	return nil
+}
+
+func (l *liquidLogic) BuildServiceInfo(_ context.Context) (liquid.ServiceInfo, error) {
+	return liquid.ServiceInfo{
+		Version: 1,
+		Resources: map[liquid.ResourceName]liquid.ResourceInfo{
+			"datacenter": {
+				HasCapacity: false,
+				HasQuota:    true,
+				Topology:    liquid.FlatResourceTopology,
+			},
+			"domain": {
+				HasCapacity: false,
+				HasQuota:    true,
+				Topology:    liquid.FlatResourceTopology,
+			},
+			"member": {
+				HasCapacity: false,
+				HasQuota:    true,
+				Topology:    liquid.FlatResourceTopology,
+			},
+			"monitor": {
+				HasCapacity: false,
+				HasQuota:    true,
+				Topology:    liquid.FlatResourceTopology,
+			},
+			"pool": {
+				HasCapacity: false,
+				HasQuota:    true,
+				Topology:    liquid.FlatResourceTopology,
+			},
+		},
+		Rates:                  map[liquid.RateName]liquid.RateInfo{},
+		CapacityMetricFamilies: map[liquid.MetricName]liquid.MetricFamilyInfo{},
+	}, nil
+}
+
+func (l *liquidLogic) ScanCapacity(ctx context.Context, req liquid.ServiceCapacityRequest, serviceInfo liquid.ServiceInfo) (liquid.ServiceCapacityReport, error) {
+	return liquid.ServiceCapacityReport{InfoVersion: 1}, nil
+}
+
+func (l *liquidLogic) ScanUsage(ctx context.Context, projectUUID string, req liquid.ServiceUsageRequest, serviceInfo liquid.ServiceInfo) (liquid.ServiceUsageReport, error) {
+	params := administrative.NewGetQuotasProjectIDParams().WithDefaults().WithContext(ctx)
+	params.ProjectID = projectUUID
+	resp, err := l.andromedaClient.Administrative.GetQuotasProjectID(params)
+	if err != nil {
+		return liquid.ServiceUsageReport{}, err
+	}
+	return liquid.ServiceUsageReport{
+		InfoVersion: 1,
+		Resources: map[liquid.ResourceName]*liquid.ResourceUsageReport{
+			"datacenter": {
```

Comment:
We recommend naming resources in plural form, e.g. `datacenters` instead of `datacenter`, because they represent the totality of all such assets within a single project.

Reply:
Andromeda uses singular names throughout all entities (inspired by openstack api) and the quota api.

On a scale from 1-10, how important would you count this recommendation? I don't want to block it, just try to understand the rationale.
Isn't the name aliased to some localized version (which could include plural)?

Reply:
It would make you stick out like a sore thumb among all the other Limes resources, because our existing liquids do in fact also fix the names for Neutron and Octavia in the same way. Since it is easy to fix this right now, I'd recommend to do it.

Reply:
In fact, here you go: https://github.com/sapcc/andromeda/pull/678